src/bin/pgcopydb/pgsql.c

```diff
@@ -2023,7 +2021,7 @@ pgsql_sync_pipeline(PGSQL *pgsql)
 
 	(void) pgsql_handle_notifications(pgsql);
 
-	while (!syncReceived)
```

Comment:
I can see why an automated tooling would reason that it is the same thing, but I prefer the extra safety check. Also communicates our intent better.

Can we have cppcheck discard that change?

Reply:
The point here is that `syncReceived` is not read at all after being set to `true`, there is a `break` statement which follows.

Reply:
I have added a comment instead.

Reply:
If the code gets refactored, it's too easy to miss the loop stop condition. I would rather use `continue` instead of `break` if you think we really do need to use the variable.